WordPressKit/StatsServiceRemoteV2.swift

```diff
@@ -279,6 +289,26 @@ extension StatsServiceRemoteV2 {
 
 }
 
+// MARK: - Mark referrer as spam helpers
+
+private extension StatsServiceRemoteV2 {
+    func pathForToggleSpamStateEndpoint(referrerDomain: String, markAsSpam: Bool) -> String {
+        let action = markAsSpam ? "new" : "delete"
+        return self.path(forEndpoint: "sites/\(siteID)/stats/referrers/spam/\(action)?domain=\(referrerDomain)", withVersion: ._1_1)
+    }
+
+    struct MarkAsSpamResponse {
```

Comment:
Nice approach, How do you feel about reducing the boiler plate by adhering this to Decodable?

Reply:
That was my intention, though `POST` method doesn't return `Data`, it already returns a dictionary, so I think that would just introduce unneeded double conversion, if I'm not mistaken :) (from dictionary to `Data`, and then decode to struct).

Reply:
Makes sense. For larger objects I think this conversion makes sense but since it's small I don't think it's required.